ui/pages/confirmation/util.ts

```diff
@@ -0,0 +1,127 @@
+import { ResultComponent } from '@metamask/approval-controller';
+
+type TemplateRendererComponent = {
+  key: string;
+  element: string;
+  props?: Record<string, unknown>;
+  children?:
+    | string
+    | TemplateRendererComponent
+    | (string | TemplateRendererComponent)[];
+};
+
+export function processError(
```

Comment:
This was me being lazy in the POC, but can we add JSDoc for all these public methods?

Reply:
Added JSDoc to the public ones.